src/Val.cc

```diff
@@ -2984,20 +2985,46 @@ void RecordVal::Assign(int field, ValPtr new_val) {
 }
 
 void RecordVal::Remove(int field) {
-    bool was_set = record_val[field].IsSet();
-    record_val[field].Reset();
+    auto& fv = record_val[field];
+    bool was_set = fv.HoldsZVal() || fv.HoldsFieldCallback();
+
+    fv.Reset();
 
     if ( was_set )
         Modified();
 }
 
+ValPtr RecordVal::GetFieldSlow(const RecordType& rt, ZValElement& fv, int field) const {
```

Comment:
This name seems a bit strange (the "slow" part is non-obvious). Off the top of my head, the best alternative that comes to mind is `GetFieldFullyResolved`.